### PR TITLE
#1636 Get rid of Lombok in langchain4j-core

### PR DESCRIPTION
**Pull request by kpavlov**
## Issue

Contributes to #1636 

## Change

Get rid of Lombok in langchain4j-core: Run Delombok refactoring in IntelliJ IDEA to remove Lombok annotations and replace them with the equivalent Java code.

This pull request focuses on removing the Lombok dependency and replacing it with manually implemented builder patterns across several classes. Additionally, it includes some minor code improvements.

## General checklist

- [x] There are no breaking changes
- [ ] I have added unit and integration tests for my change
- [ ] I have manually run all the unit tests in all modules, and they are all green
- [ ] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for adding new maven module

- [ ] I have added my new module in the root `pom.xml` and `langchain4j-bom/pom.xml`

## Checklist for adding new embedding store integration

- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreRemovalIT` that extends from `EmbeddingStoreWithRemovalIT`

## Checklist for changing existing embedding store integration

- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

File: langchain4j-core/src/test/java/dev/langchain4j/agent/tool/ToolSpecificationsTest.java [hunk 1]
```diff
@@ -1,39 +1,26 @@
...
 
-import dev.langchain4j.model.chat.request.json.JsonArraySchema;
-import dev.langchain4j.model.chat.request.json.JsonBooleanSchema;
-import dev.langchain4j.model.chat.request.json.JsonEnumSchema;
-import dev.langchain4j.model.chat.request.json.JsonIntegerSchema;
-import dev.langchain4j.model.chat.request.json.JsonNumberSchema;
-import dev.langchain4j.model.chat.request.json.JsonObjectSchema;
-import dev.langchain4j.model.chat.request.json.JsonSchemaElement;
-import dev.langchain4j.model.chat.request.json.JsonStringSchema;
+import dev.langchain4j.model.chat.request.json.*;
```
Comment (glaforge) [inline langchain4j-core/src/test/java/dev/langchain4j/agent/tool/ToolSpecificationsTest.java]: Try to keep individual imports, and avoid using star imports.

Comment (kpavlov) [reply langchain4j-core/src/test/java/dev/langchain4j/agent/tool/ToolSpecificationsTest.java]: Sorry, missed that. Will add .editorconfig

**Review by glaforge — DISMISSED** [review]
Those changes look great to me!

**Comment by kpavlov** [comment]
Looking forward for having Codacy or other quality enforcement tool to be integrated:)

File: langchain4j-core/src/main/java/dev/langchain4j/rag/DefaultRetrievalAugmentor.java [hunk 2]
```diff
@@ -264,12 +272,16 @@ private static void log(Query query, ContentRetriever retriever, List<Content> c
         log.debug("Retrieved {} contents using query '{}' and retriever '{}'",
                 contents.size(), query.text(), retriever);
 
-        if (contents.size() > 0) {
+        if (!contents.isEmpty()) {
```
Comment (Wadimz) [inline langchain4j-core/src/main/java/dev/langchain4j/rag/DefaultRetrievalAugmentor.java:277]: can we add here ` && log.isTraceEnabled()` to not build contentsString when logging was not requested?

**Review by Wadimz — APPROVED** [review]
Thanks for format and `*` import unwrapping!

**Review by Wadimz — APPROVED** [review]

File: langchain4j-core/src/main/java/dev/langchain4j/rag/DefaultRetrievalAugmentor.java [hunk 3]
```diff
@@ -227,62 +225,69 @@ private static List<Content> retrieve(ContentRetriever retriever, Query query) {
...
                     .map(segment -> "- " + escapeNewlines(segment.text()))
                     .collect(joining("\n"));
-            log.trace("Retrieved {} contents using query '{}' and retriever '{}':\n{}",
+                log.trace("Retrieved {} contents using query '{}' and retriever '{}':\n{}",
                     contents.size(),
                     query.text(),
                     retriever.getClass().getName(),
                     contentsSting);
+            }
+        } else {
```
Comment (Wadimz) [inline langchain4j-core/src/main/java/dev/langchain4j/rag/DefaultRetrievalAugmentor.java:287]: looks like else belongs to wrong if branch here. Should it be for inner if?

File: langchain4j-core/src/main/java/dev/langchain4j/rag/DefaultRetrievalAugmentor.java [hunk 4]
```diff
@@ -227,62 +225,69 @@ private static List<Content> retrieve(ContentRetriever retriever, Query query) {
...
     private static void logQueries(Query originalQuery, Collection<Query> queries) {
         if (queries.size() == 1) {
             Query transformedQuery = queries.iterator().next();
             if (!transformedQuery.equals(originalQuery)) {
                 log.debug("Transformed original query '{}' into '{}'",
-                        originalQuery.text(), transformedQuery.text());
+                    originalQuery.text(), transformedQuery.text());
             }
-        } else {
+        } else if (log.isDebugEnabled()){
```
Comment (Wadimz) [inline langchain4j-core/src/main/java/dev/langchain4j/rag/DefaultRetrievalAugmentor.java:246]: if we started updating logging, maybe we can wrap a whole content of this method with `if (log.isDebugEnabled())`. Looks all it does is logs with debug level

Comment (kpavlov) [reply langchain4j-core/src/main/java/dev/langchain4j/rag/DefaultRetrievalAugmentor.java:246]: Let's stop updating logging in this PR. Let's do it separately
Things are getting messy

**Review by Wadimz — APPROVED** [review]

File: langchain4j-core/src/main/java/dev/langchain4j/rag/DefaultRetrievalAugmentor.java (hunk 3 above)
Comment (kpavlov) [reply langchain4j-core/src/main/java/dev/langchain4j/rag/DefaultRetrievalAugmentor.java:287]: [Fixed](https://github.com/langchain4j/langchain4j/pull/1987/files#diff-9a5519a26a4b6d2fd412c877de4459c2a119e793fece9385f558a93a7e0aee5aR273-R275)

**Review by Wadimz — APPROVED** [review]

File: langchain4j-core/src/main/java/dev/langchain4j/rag/DefaultRetrievalAugmentor.java [hunk 5]
```diff
@@ -219,58 +225,71 @@ private static List<Content> retrieve(ContentRetriever retriever, Query query) {
...
-                    contents.size(), query.text(), retriever, contents.stream()
-                            .map(Content::textSegment)
-                            .map(segment -> "- " + escapeNewlines(segment.text()))
-                            .collect(joining("\n")));
+                contents.size(),
+                query.text(),
+                retriever.getClass().getName(),
+                contentsSting);
+        } else {
+            log.trace("Retrieved empty contents using query '{}' and retriever '{}'",
```
Comment (langchain4j) [inline langchain4j-core/src/main/java/dev/langchain4j/rag/DefaultRetrievalAugmentor.java]:
```suggestion
            log.trace("Retrieved 0 contents using query '{}' and retriever '{}'",
```

**Review by ghost — DISMISSED** [review]
@katcharov thank you!

**Comment by langchain4j** [comment]
@Wadimz @glaforge thank you for reviewing!

**Pull request merged**